src/uploader.ts

```diff
@@ -31,7 +32,7 @@ export default class Uploader {
 
     this.client = new S3(this.options.clientConfig)
 
-    this.directory = this.options.directory ? this.options.directory : `${this.vite.root}/${this.vite.build.outDir}`
+    this.directory = this.options.directory ? this.options.directory : normalize(`${this.vite.root}/${this.vite.build.outDir}`)
```

Comment:
I found that `path.resolve` works better in this case, because it adds ability to work with absolute paths. `this.vite.build.outDir` also supports absolute paths, so this behavior will be more predictable.

PR is updated.